### ISSUE TITLE
Implement RMSProp Feature

### ISSUE THREAD

**Issue by jroose** [jroose/Axono#2]
Must implement the RMSProp adaptive learning rate algorithm.  The implementation should be generic enough to be applied to any Delta strategy.  The algorithm should be implemented as part of a generic LearningRate strategy responsible for fine-tuning learning rates for individual neurons.  Global learning rates should be re-implemented as an alternative GlobalLearningRate strategy.
